.github/workflows/hack.yml

```diff
@@ -0,0 +1,35 @@
+name: Hack
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  hack:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo
+            target
+          key: ${{ matrix.os }}-hack
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        run: |
+          sudo apt install -y libboost-dev
+          cargo install cargo-hack --locked
+
+      - name: Hack Check
+        # TODO: remove the exclude when `cuprate-database` errors are fixed
+        run: cargo hack --workspace --exclude cuprate-database check --feature-powerset --no-dev-deps
```

Comment:
`cuprate-database` does not build with `--no-default-features` and I don't think there is a way to fix this without hot swapable backends, so another reason for that :)

Reply:
Should we set `cuprate-database` to always pull in `heed` by default? Then if another backend feature is detected, that is used instead, although `heed` will still be built (but not used).

Reply:
We could, initially I didn't want to, but thinking about it more, with hot-swapable backends both `heed` & `redb` would need to be pulled in all the time as well.

I'll add this